File: 0352-data-stream-as-disjoint-intervals/0352-data-stream-as-disjoint-intervals.cs
public class SummaryRanges
{
    private readonly SortedSet<int> _integersSeen;

    public SummaryRanges()
    {
        _integersSeen = new SortedSet<int>();
    }

    public void AddNum(int value)
    {
        _integersSeen.Add(value);
    }

    public int[][] GetIntervals()
    {
        if (_integersSeen.Count == 0) return Array.Empty<int[]>();

        var intervals = new List<int[]>();
        var (left, right) = (-1, -1);

        foreach (var num in _integersSeen)
        {
            if (left < 0)
            {
                left = right = num;
            } else if (num == right + 1)
            {
                right = num;
            }
            else
            {
                intervals.Add(new int[] {left, right});
                left = right = num;
            }
        }
        intervals.Add(new int[] {left, right});
        return intervals.ToArray();
    }
}

/**
 * Your SummaryRanges object will be instantiated and called as such:
 * SummaryRanges obj = new SummaryRanges();
 * obj.AddNum(value);
 * int[][] param_2 = obj.GetIntervals();
 */

File: 0352-data-stream-as-disjoint-intervals/0352-data-stream-as-disjoint-intervals.py
from sortedcontainers import SortedSet

class SummaryRanges:
    
    def __init__(self):
        self.__integers_seen = SortedSet()

    def addNum(self, value: int) -> None:
        self.__integers_seen.add(value)

    def getIntervals(self) -> List[List[int]]:
        if not self.__integers_seen: return [[]]
        
        intervals = [];
        left = right = -1
        
        for value in self.__integers_seen:
            if left < 0:
                left = right = value
            elif value == right + 1:
                right = value
            else:
                intervals.append([left, right])
                left = right = value
        
        intervals.append([left, right])
        return intervals


# Your SummaryRanges object will be instantiated and called as such:
# obj = SummaryRanges()
# obj.addNum(value)
# param_2 = obj.getIntervals()